### PR TITLE
Bdd/api setup - entities and seeding

### PR DESCRIPTION
**Pull request by teasmade**
- Add base entities
- Add basic resolvers to allow for schema build
- Add seed data in .json format
- Add seeder script for base entities, modifier assignments and some test combats
- Break out modifier assignment and character stat generation logic into seeder util methods
- Clean data and code for variable / parameter naming (notably use of "assignment" (EN) vs. "assignement" (FR)

### CODE REVIEW

**Review by julien44830 — APPROVED** [review]

File: api/src/competitor/competitor.entity.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  @Field()
+  @UpdateDateColumn()
+  updatedAt: Date;
+
+  @Field(() => Profession)
+  @ManyToOne(() => Profession, (profession) => profession)
+  profession: Profession;
+
+  @Field(() => Image)
+  @ManyToOne(() => Image, (id_avatar_image) => id_avatar_image.id)
```
Comment (PierreDelarocque) [inline api/src/competitor/competitor.entity.ts]: change the underscord to camelcase for id_avatar_image_.id

File: api/src/image/image.entity.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  @Field()
+  @Column()
+  path: string;
+
+  @Field(() => God)
+  @OneToMany(() => God, (god) => god.image)
+  god: God[];
+
+  @Field(() => Competitor)
+  @OneToMany(() => Competitor, (id_avatar_image) => id_avatar_image.id)
```
Comment (PierreDelarocque) [inline api/src/image/image.entity.ts]: change the underscord to camelcase for id_avatar_image_.id

File: api/src/image/image.entity.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  idAvatarImage: Competitor[];
+
+  @Field(() => Competitor)
+  @OneToMany(() => Competitor, (id_battle_image) => id_battle_image.id)
```
Comment (PierreDelarocque) [inline api/src/image/image.entity.ts]: change the underscord to camelcase for id_avatar_image_.id

**Review by PierreDelarocque — CHANGES_REQUESTED** [review]
change the underscord for a camelCase, trial_id an id_avatar_image.id for harmonization

File: api/src/db/seed-data/image.json [hunk 4]
```diff
@@ -0,0 +1,200 @@
+[
+  {
+    "type": "god",
+    "target_entity": "Aphrodite",
+    "path": "/img/gods/aphrodite.png",
+    "id": 0
```
Comment (julien44830) [inline api/src/db/seed-data/image.json]: why id 0 in all objet
` "id": 0`

**Review by julien44830 — CHANGES_REQUESTED** [review]

**Review by Gadln — DISMISSED** [review]

File: api/src/db/seed-data/image.json (hunk 4 above)
Comment (teasmade) [reply api/src/db/seed-data/image.json]: Yes, this is no longer needed

File: api/src/image/image.entity.ts (hunk 3 above)
Comment (teasmade) [reply api/src/image/image.entity.ts]: Done

**Review by julien44830 — DISMISSED** [review]

File: api/src/image/image.entity.ts (hunk 2 above)
Comment (teasmade) [reply api/src/image/image.entity.ts]: done

File: api/src/competitor/competitor.entity.ts (hunk 1 above)
Comment (teasmade) [reply api/src/competitor/competitor.entity.ts]: done

**Review by julien44830 — DISMISSED** [review]

**Review by Adri-bis — DISMISSED** [review]

**Review by julien44830 — DISMISSED** [review]

**Review by PierreDelarocque — DISMISSED** [review]

File: api/src/db/seed.ts [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+          newModifierAssignment.modifiedEntityId = combat.id;
+          newModifierAssignment.modifiedEntityType = "combat";
+          return await newModifierAssignment.save();
+        })
+      )
+    );
+
+    // COMMIT!
+    await queryRunner.commitTransaction();
+    console.info(
```
Comment (jujuck) [inline api/src/db/seed.ts]: Est ce vraiment lisible dans le terminal ?

File: api/src/index.ts [hunk 6]
```diff
@@ -5,14 +5,23 @@ import { appDataSource } from "./db/db-client";
...
 
 (async () => {
   await appDataSource.initialize();
   const schema = await buildSchema({
-    resolvers: [GodResolver],
+    resolvers: [
+      GodResolver,
+      CompetitorResolver,
+      ProfessionResolver,
+
```
Comment (jujuck) [inline api/src/index.ts]: Ligne morte

File: api/src/db/seed.utils.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    return values;
+  }
+
+  const statDistribution = generateRandomArray(
+    totalStatsBonus,
+    possiblePlayerModifiers.length,
+    maxStatBonus
+  );
+
+  const modifierAssignments = [];
```
Comment (jujuck) [inline api/src/db/seed.utils.ts]: Pourquoi ne pas mettre un .map ?

File: api/src/db/seed.utils.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+   * @param parts The number of parts to divide the sum into.
+   * @param maxValue The maximum value for each part.
+   * @returns An array of randomly ordered values that always add up to the given sum value.
+   */
+  function generateRandomArray(
+    sum: number,
+    parts: number,
+    maxValue: number
+  ): number[] {
+    const values = [];
```
Comment (jujuck) [inline api/src/db/seed.utils.ts]: Typage

**Review by jujuck — CHANGES_REQUESTED** [review]

File: api/src/db/seed.utils.ts (hunk 8 above)
Comment (teasmade) [reply api/src/db/seed.utils.ts]: Done

File: api/src/db/seed.utils.ts (hunk 7 above)
Comment (teasmade) [reply api/src/db/seed.utils.ts]: Yes c'est plus propre à typer avec un .map

File: api/src/db/seed.ts (hunk 5 above)
Comment (teasmade) [reply api/src/db/seed.ts]: Effectivement on peut faire mieux...

image

File: api/src/index.ts (hunk 6 above)
Comment (teasmade) [reply api/src/index.ts]: Done

**Review by Adri-bis — APPROVED** [review]

**Review by julien44830 — APPROVED** [review]

**Review by PierreDelarocque — APPROVED** [review]

**Review by Gadln — APPROVED** [review]

**Review by jujuck — APPROVED** [review]

**Pull request merged**